### DISCUSSION TITLE
Issue: dify-api starts very slowly

### DISCUSSION THREAD

**Discussion by LouisCan** [langgenius/dify#10346] [General]
### Self Checks

- [X] I have searched for existing issues [search for existing issues](https://github.com/langgenius/dify/issues), including closed ones.
- [X] I confirm that I am using English to submit this report (我已阅读并同意 [Language Policy](https://github.com/langgenius/dify/issues/1542)).
- [X] [FOR CHINESE USERS] 请务必使用英文提交 Issue，否则会被关闭。谢谢！:）
- [X] Please do not modify this template :) and fill in all the required fields.

### Content

Issue: dify-api starts very slowly

docker logs -f docker-api-1
Running migrations
None of PyTorch, TensorFlow >= 2.0, or Flax have been found. Models won't be available and only tokenizers, configuration and file/data utilities can be used.
sagemaker.config INFO - Not applying SDK defaults from location: /etc/xdg/sagemaker/config.yaml
sagemaker.config INFO - Not applying SDK defaults from location: /root/.config/sagemaker/config.yaml
2024-11-06 02:51:43,688.688 INFO [pre_load_builtin_providers_cache] [font_manager.py:1578] - generated new fontManager
Preparing database migration...
Database migration skipped
None of PyTorch, TensorFlow >= 2.0, or Flax have been found. Models won't be available and only tokenizers, configuration and file/data utilities can be used.
sagemaker.config INFO - Not applying SDK defaults from location: /etc/xdg/sagemaker/config.yaml
sagemaker.config INFO - Not applying SDK defaults from location: /root/.config/sagemaker/config.yaml

**Comment by mayun8172** [comment]
Any advice on this to debug this problem? @crazywoola

**Comment by bikevit2008** [comment]
I have this issue together, and i don't know what i must to do, how to fix it?

```sh
root@cv4338665:~/dify/docker# docker logs docker-api-1
Running migrations
None of PyTorch, TensorFlow >= 2.0, or Flax have been found. Models won't be available and only tokenizers, configuration and file/data utilities can be used.
/app/api/.venv/lib/python3.12/site-packages/tencentcloud/hunyuan/v20230901/models.py:5585: SyntaxWarning: invalid escape sequence '\_'
  """function名称，只能包含a-z，A-Z，0-9，\_或-
/app/api/.venv/lib/python3.12/site-packages/jieba/__init__.py:44: SyntaxWarning: invalid escape sequence '\.'
  re_han_default = re.compile("([\u4E00-\u9FD5a-zA-Z0-9+#&\._%\-]+)", re.U)
/app/api/.venv/lib/python3.12/site-packages/jieba/__init__.py:46: SyntaxWarning: invalid escape sequence '\s'
  re_skip_default = re.compile("(\r\n|\s)", re.U)
/app/api/.venv/lib/python3.12/site-packages/jieba/finalseg/__init__.py:78: SyntaxWarning: invalid escape sequence '\.'
  re_skip = re.compile("([a-zA-Z0-9]+(?:\.\d+)?%?)")
/app/api/.venv/lib/python3.12/site-packages/jieba/posseg/__init__.py:16: SyntaxWarning: invalid escape sequence '\.'
  re_skip_detail = re.compile("([\.0-9]+|[a-zA-Z0-9]+)")
/app/api/.venv/lib/python3.12/site-packages/jieba/posseg/__init__.py:17: SyntaxWarning: invalid escape sequence '\.'
  re_han_internal = re.compile("([\u4E00-\u9FD5a-zA-Z0-9+#&\._]+)")
/app/api/.venv/lib/python3.12/site-packages/jieba/posseg/__init__.py:18: SyntaxWarning: invalid escape sequence '\s'
  re_skip_internal = re.compile("(\r\n|\s)")
/app/api/.venv/lib/python3.12/site-packages/jieba/posseg/__init__.py:21: SyntaxWarning: invalid escape sequence '\.'
  re_num = re.compile("[\.0-9]+")
2024-12-13 15:12:59.117 INFO [pre_load_builtin_providers_cache] [font_manager.py:1578] - generated new fontManager
```

**Reply by bikevit2008** [reply]
And nothing after, on arm mac os i have 

```sh
2024-12-13 15:16:37 None of PyTorch, TensorFlow >= 2.0, or Flax have been found. Models won't be available and only tokenizers, configuration and file/data utilities can be used.
2024-12-13 15:16:53 None of PyTorch, TensorFlow >= 2.0, or Flax have been found. Models won't be available and only tokenizers, configuration and file/data utilities can be used.
2024-12-13 15:17:00 [2024-12-13 12:17:00 +0000] [1] [INFO] Starting gunicorn 22.0.0
2024-12-13 15:17:00 [2024-12-13 12:17:00 +0000] [1] [INFO] Listening at: http://0.0.0.0:5001 (1)
2024-12-13 15:17:00 [2024-12-13 12:17:00 +0000] [1] [INFO] Using worker: gevent
2024-12-13 15:17:00 [2024-12-13 12:17:00 +0000] [79] [INFO] Booting worker with pid: 79
```

**Reply by bikevit2008** [reply]
dify on cloud amd64 server with my issue
CONTAINER ID   NAME                  CPU %     MEM USAGE / LIMIT     MEM %     NET I/O           BLOCK I/O        PIDS
f7553c694fde   docker-nginx-1        0.00%     4.789MiB / 7.755GiB   0.06%     4kB / 2.74kB      0B / 16.4kB      6
a911fb4231f5   docker-api-1          0.00%     659.5MiB / 7.755GiB   8.30%     3.81kB / 1.05kB   164kB / 140MB    7
a1cfa0001313   docker-worker-1       0.00%     657.4MiB / 7.755GiB   8.28%     3.77kB / 1.05kB   16.4kB / 141MB   7
a6dd9515d314   docker-web-1          0.31%     58.45MiB / 7.755GiB   0.74%     3.38kB / 1.97kB   0B / 28.7kB      34
9403ea19cea4   docker-db-1           2.18%     24.21MiB / 7.755GiB   0.30%     1.78kB / 0B       741kB / 53.5MB   6
656fffd71d6d   docker-redis-1        0.10%     3.727MiB / 7.755GiB   0.05%     1.78kB / 0B       1.42MB / 0B      5
9dad5f15205c   docker-sandbox-1      0.00%     35.51MiB / 7.755GiB   0.45%     2.93kB / 1.01kB   1.75MB / 189MB   8
7dad90d3c2ec   docker-ssrf_proxy-1   0.01%     13.12MiB / 7.755GiB   0.17%     4.28kB / 1.33kB   0B / 885kB       9
4219ab505c7b   docker-weaviate-1     0.05%     25.21MiB / 7.755GiB   0.32%     1.89kB / 0B       0B / 86kB        9

**Reply by bikevit2008** [reply]
My docker info with issue

root@cv4338665:~/dify/docker# docker info
Client: Docker Engine - Community
 Version:    27.4.0
 Context:    default
 Debug Mode: false
 Plugins:
  buildx: Docker Buildx (Docker Inc.)
    Version:  v0.19.2
    Path:     /usr/libexec/docker/cli-plugins/docker-buildx
  compose: Docker Compose (Docker Inc.)
    Version:  v2.31.0
    Path:     /usr/libexec/docker/cli-plugins/docker-compose

Server:
 Containers: 9
  Running: 9
  Paused: 0
  Stopped: 0
 Images: 8
 Server Version: 27.4.0
 Storage Driver: overlay2
  Backing Filesystem: extfs
  Supports d_type: true
  Using metacopy: false
  Native Overlay Diff: true
  userxattr: false
 Logging Driver: json-file
 Cgroup Driver: systemd
 Cgroup Version: 2
 Plugins:
  Volume: local
  Network: bridge host ipvlan macvlan null overlay
  Log: awslogs fluentd gcplogs gelf journald json-file local splunk syslog
 Swarm: inactive
 Runtimes: io.containerd.runc.v2 runc
 Default Runtime: runc
 Init Binary: docker-init
 containerd version: 88bf19b2105c8b17560993bee28a01ddc2f97182
 runc version: v1.2.2-0-g7cb3632
 init version: de40ad0
 Security Options:
  apparmor
  seccomp
   Profile: builtin
  cgroupns
 Kernel Version: 6.8.0-50-generic
 Operating System: Ubuntu 24.04.1 LTS
 OSType: linux
 Architecture: x86_64
 CPUs: 4
 Total Memory: 7.755GiB
 Name: cv4338665.novalocal
 ID: 442d4f3b-3e2f-435a-a749-a69c57861099
 Docker Root Dir: /var/lib/docker
 Debug Mode: false
 Experimental: false
 Insecure Registries:
  127.0.0.0/8
 Live Restore Enabled: false

WARNING: bridge-nf-call-iptables is disabled
WARNING: bridge-nf-call-ip6tables is disabled
